### PR TITLE
feat: Updated Azure.APIM.AvailabilityZone

### LINKED ISSUE #2788: [FEATURE] Add rule for APIM to check capacity is set to same or more than number of zones deployed too
### Your suggestion

Add rule for APIM to check capacity is set to same or more than number of zones deployed too as per guidance here: https://learn.microsoft.com/en-us/azure/api-management/high-availability?toc=%2Fazure%2Freliability%2Ftoc.json&bc=%2Fazure%2Freliability%2Fbreadcrumb%2Ftoc.json#:~:text=When%20you%20enable,hosts%20one%20unit.

### Alternatives

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2788

Updated Azure.APIM.AvailabilityZone.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
Ready for review @BernieWhite. I would appreciate your feedback on it.

File: docs/CHANGELOG-v1.md [hunk 1]
```diff
@@ -39,6 +39,14 @@ See [upgrade notes][1] for helpful information when upgrading from previous vers
       [#2846](https://github.com/Azure/PSRule.Rules.Azure/issues/2846)
     - Check that database accounts have public network access disabled by @BenjaminEngeset.
       [#2702](https://github.com/Azure/PSRule.Rules.Azure/issues/2702)
+- Updated rules:
+  - API Management:
+    - Removed the `If` Premium SKU check.
+    - Added check for Premium SKU.
+    - Updated tests.
+    - Updated rule doc.
+      [#2788](https://github.com/Azure/PSRule.Rules.Azure/issues/2788)
```
Comment (BernieWhite) [inline docs/CHANGELOG-v1.md:48]:
```suggestion
  - API Management:
    - **Important change**: Updated `Azure.APIM.AvailabilityZone` to improve accuracy with non-premium SKUs by @BenjaminEngeset.
      [#2788](https://github.com/Azure/PSRule.Rules.Azure/issues/2788)
      - Removed the `If` Premium SKU.
      - Added check for Premium SKU.
      - Updated tests.
      - Updated rule doc.
      - Bumped rule set to `2024_06`.
```

File: docs/en/rules/Azure.APIM.AvailabilityZone.md [hunk 2]
```diff
@@ -173,6 +176,10 @@ resource service_api_mgmt_test2_name_resource 'Microsoft.ApiManagement/service@2
 }
 ```
 
+## NOTES
+
+For developer environments, supressing the rule might make sense as enabling zone redundancy for an API Management instance requries the `Premium` SKU currently.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.APIM.AvailabilityZone.md:181]: The `NOTES` header is listed above. I like your positioning of `NOTES`, so maybe move the content from the `NOTES` above to this section so there is only one `NOTES`.

```suggestion
For developer environments, suppressing the rule might make sense as enabling zone redundancy for an API Management instance requries the `Premium` SKU currently.
```

File: src/PSRule.Rules.Azure/rules/Azure.APIM.Rule.ps1 [hunk 3]
```diff
@@ -193,8 +193,14 @@ Rule 'Azure.APIM.CertificateExpiry' -Ref 'AZR-000051' -Type 'Microsoft.ApiManage
     }
 } -Configure @{ Azure_MinimumCertificateLifetime = 30 }
 
-# Synopsis: API management services deployed with Premium SKU should use availability zones in supported regions for high availability.
-Rule 'Azure.APIM.AvailabilityZone' -Ref 'AZR-000052' -Type 'Microsoft.ApiManagement/service' -If { IsPremiumAPIM } -Tag @{ release = 'GA'; ruleSet = '2021_12'; 'Azure.WAF/pillar' = 'Reliability'; } {
+# Synopsis: API Management instances should use availability zones in supported regions for high availability.
+Rule 'Azure.APIM.AvailabilityZone' -Ref 'AZR-000052' -Type 'Microsoft.ApiManagement/service' -Tag @{ release = 'GA'; ruleSet = '2021_12'; 'Azure.WAF/pillar' = 'Reliability'; } {
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.APIM.Rule.ps1:197]: Let's bump the rule set as the behaviour has changed. You may need to update some baselines in baseline tests as a result.

```suggestion
Rule 'Azure.APIM.AvailabilityZone' -Ref 'AZR-000052' -Type 'Microsoft.ApiManagement/service' -Tag @{ release = 'GA'; ruleSet = '2024_06'; 'Azure.WAF/pillar' = 'Reliability'; } {
```

File: docs/en/rules/Azure.APIM.AvailabilityZone.md [hunk 4]
```diff
@@ -6,26 +6,29 @@ resource: API Management
...
+API Management instances using availability zones improve reliability and ensure availability during failure scenarios affecting a data center within a region.
+
+Enabling zone redundancy for an API Management instance in a primary region provides redundancy for all service components: gateway, management plane, and developer portal are replicated across data centers in physically separated zones, making it resilient to a zone failure.
+
+Enabling zone redundancy for an API Management instance in a additional region provides redundancy for *only* the gateway service component.
 
 ## RECOMMENDATION
 
-Consider using availability zones for API management services deployed with Premium SKU.
+Consider using availability zones for API Management instances.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.APIM.AvailabilityZone.md:25]:
```suggestion
Consider using at least two (2) scale units and availability zones to improve resiliency of API Management instances to zone failures.
```

File: docs/en/rules/Azure.APIM.AvailabilityZone.md [hunk 5]
```diff
@@ -6,26 +6,29 @@ resource: API Management
...
 
 ## DESCRIPTION
 
-API management services using availability zones improve reliability and ensure availability during failure scenarios affecting a data center within a region.
-With zone redundancy, the gateway and the control plane of your API Management instance (Management API, developer portal, Git configuration) are replicated across data centers in physically separated zones, making it resilient to a zone failure.
+API Management instances using availability zones improve reliability and ensure availability during failure scenarios affecting a data center within a region.
+
+Enabling zone redundancy for an API Management instance in a primary region provides redundancy for all service components: gateway, management plane, and developer portal are replicated across data centers in physically separated zones, making it resilient to a zone failure.
+
+Enabling zone redundancy for an API Management instance in a additional region provides redundancy for *only* the gateway service component.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.APIM.AvailabilityZone.md:21]:
```suggestion
API Management supports using availability zones to provide zone redundency for key components.
Zone redundancy improves resiliency and high availability of the service by deploying scale units across data centers in physically separated zones.

The following are required to deploy a zone redundant configuration:

- API Management is deployed with the Premium SKU.
- At least two scale units must be deployed in separate availbility zones

Zone redundancy has a different configuration for the primary and additional regions:

- In the primary region, all service components are replicated across zones (gateway, management plane, and developer portal).
- In additional regions, *only* the gateway is replicated across zones.
  As the management plane/ developer portal service components are only deployed to the primary region.
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Thanks @BenjaminEngeset. Great work. Please see my suggestions below.

**Comment by BenjaminEngeset** [comment]
I think we should be all set, @BernieWhite. Please check the latest updates.

**Review by BernieWhite — APPROVED** [review]
Great work @BenjaminEngeset. All good to merge.

**Pull request merged**